src/main/java/com/rocketden/main/service/GameManagementService.java

```diff
@@ -91,9 +98,28 @@ public RoomDto startGame(String roomId, StartGameRequest request) {
     // Initialize and add a game object from a room object
     public void createAddGameFromRoom(Room room) {
         Game game = GameMapper.fromRoom(room);
+        game.setProblems(getProblemsFromDifficulty(room.getDifficulty(), 1));
         currentGameMap.put(room.getRoomId(), game);
     }
 
+    // Choose the problem based on problem difficulty settings (empty list for invalid request)
+    private List<Problem> getProblemsFromDifficulty(ProblemDifficulty difficulty, int n) {
+        ProblemSettingsDto request = new ProblemSettingsDto();
+        request.setDifficulty(difficulty);
+        request.setN(n);
+
+        try {
+            List<Problem> problems = new ArrayList<>();
+            for (ProblemDto problemDto : problemService.getRandomProblems(request)) {
+                problems.add(ProblemMapper.toEntity(problemDto));
```

Comment:
Now that we no longer need to call `getRandomProblems` from a GET request, maybe we could change it to return `Problem` instead of `ProblemDto` that way we don't need this conversion?

Reply:
I still kept the GET request and that setup in `ProblemController.java` as well as the associated tests, since I could see us using that in the future, but I made the modification to return `Problem` instead of `ProblemDto`. This does require a conversion in `ProblemController.java`, which is the only one across the controllers, so let me know if anything should change there.